File: tests/machine/x/go/sum_builtin.go
//go:build ignore

package main

import (
	"fmt"
	"reflect"

	"golang.org/x/exp/constraints"
)

func main() {
	fmt.Println(_sprint(_sumOrdered[int]([]int{1, 2, 3})))
}

func _sprint(v any) string {
	if v == nil {
		return "<nil>"
	}
	rv := reflect.ValueOf(v)
	if (rv.Kind() == reflect.Map || rv.Kind() == reflect.Slice) && rv.IsNil() {
		return "<nil>"
	}
	return fmt.Sprint(v)
}

func _sumOrdered[T constraints.Integer | constraints.Float](s []T) float64 {
	var sum float64
	for _, v := range s {
		sum += float64(v)
	}
	return sum
}

File: tests/machine/x/zig/sum_builtin.zig
const std = @import("std");

fn _sum_int(v: []const i32) i32 {
    var sum: i32 = 0;
    for (v) |it| { sum += it; }
    return sum;
}

pub fn main() void {
    std.debug.print("{any}\n", .{_sum_int(&[_]i32{1, 2, 3})});
}
